internal/transport/http2_server.go

```diff
@@ -139,9 +139,10 @@ func NewServerTransport(conn net.Conn, config *ServerConfig) (_ ServerTransport,
 		var err error
 		conn, authInfo, err = config.Credentials.ServerHandshake(rawConn)
 		if err != nil {
-			// ErrConnDispatched means that the connection was dispatched away from
-			// gRPC; those connections should be left open.
-			if err == credentials.ErrConnDispatched {
+			// ErrConnDispatched means that the connection was dispatched away
+			// from gRPC; those connections should be left open. To prevent log
+			// spam, return these errors directly.
```

Comment:
Maybe....
```go
// ... should be left open.  io.EOF means the connection
// was closed before handshaking completed, which can happen
// naturally from probers.  Return these errors directly.
```

(`ErrConnDispatched` isn't returned to prevent log spam; it's returned so gRPC knows not to close the connection.)

Reply:
Yeah, that's why I originally had the two separate. Changed to your comment.